appengine/genetics/js/genetics.js

```diff
@@ -0,0 +1,403 @@
+      div.style.width = divWidth;
+    }
+  };
+  window.addEventListener('scroll', function() {
+    onresize();
+    Blockly.svgResize(BlocklyGames.workspace);
+  });
+  window.addEventListener('resize', onresize);
+  onresize();
+
+  // Handle SELECT events dispatched by tabs.
+  goog.events.listen(Genetics.tabbar, goog.ui.Component.EventType.SELECT,
+      function(e) {
+        var index = e.target.getParent().getSelectedTabIndex();
+        Genetics.changeTab(index);
+      });
+
+  // Inject JS editor.
+  var defaultCode = '';
+  BlocklyInterface.editor = window['ace']['edit']('editor');
+  BlocklyInterface.editor['setTheme']('ace/theme/chrome');
+  BlocklyInterface.editor['setShowPrintMargin'](false);
+  BlocklyInterface.editor['setOptions']({
+    'enableBasicAutocompletion': true,
+    'enableSnippets': true,
+    'enableLiveAutocompletion': false
+  });
+  var session = BlocklyInterface.editor['getSession']();
+  session['setMode']('ace/mode/javascript');
+  session['setTabSize'](2);
+  session['setUseSoftTabs'](true);
+  session['on']('change', Genetics.editorChanged);
+  BlocklyInterface.editor['setValue'](defaultCode, -1);
+
+  // Inject Blockly.
+  var toolbox = document.getElementById('toolbox');
+  BlocklyGames.workspace = Blockly.inject('blockly',
+      {
+        'media': 'third-party/blockly/media/',
+        'rtl': false,
+        'toolbox': toolbox,
+        'trashcan': true,
+        'zoom': {'controls': true, 'wheel': true}
+      });
+  // Disable blocks not within a function.
+  BlocklyGames.workspace.addChangeListener(Blockly.Events.disableOrphans);
+  Blockly.JavaScript.addReservedWords('pickFight,proposeMate,acceptMate');
+  var defaultXml =
+  '<xml>' +
+    '<block type="genetics_pickFight" deletable="false" editable="false" ' +
+        'x="0" y="150">' +
+      '<comment pinned="false">' +
+'Chooses and returns a mouse from mice to pick a fight with or null to ' +
+    'choose no\n' +
+'mouse and never fight again.\n' +
+'Choosing to fight against itself will kill the mouse.\n' +
+'@returns the mouse chosen to attempt to mate with or null if no mouse is ' +
+    'chosen' +
+      '</comment>' +
+      '<value name="RETURN">' +
+        '<shadow type="logic_null"></shadow>' +
+      '</value>' +
+    '</block>' +
+    '<block type="genetics_proposeMate" deletable="false" editable="false" ' +
+        'x="0" y="350">' +
+      '<comment pinned="false">' +
+'Chooses and returns a mouse from mice to attempt to mate with or null ' +
+    'to choose\n' +
+'no mouse and never mate again. If the mate chosen is valid and agrees to ' +
+    'the\n' +
+'request then a child will be born.\n' +
+'@returns the mouse chosen to attempt to mate with or null if no mouse is ' +
+    'chosen' +
+      '</comment>' +
+      '<value name="RETURN">' +
+        '<shadow type="logic_null"></shadow>' +
+      '</value>' +
+    '</block>' +
+    '<block type="genetics_acceptMate" deletable="false" editable="false" ' +
```

Comment:
What does editable=false do on these three blocks?  There are no fields to edit.  Is this needed?